submodules/telemetry/__init__.py

```diff
@@ -1,166 +1,129 @@
-# TODO: Uses placeholder variables in other files, so make it use actual values
-# TODO: what if radio turned off during a burst?
 import base64
 import collections
 import logging
-import struct
-import time
-import core
 from functools import partial
 from threading import Lock
+from time import sleep
 
 from core.mode import Mode
 
 from core import config
 from core.threadhandler import ThreadHandler
 from submodules import radio_output
-from submodules.command_ingest import command
-
-
-logger = logging.getLogger("TELEMETRY")
-
-telem_packet_buffer = collections.deque(
-    maxlen=config['telemetry']['buffer_size'])
-event_packet_buffer = collections.deque(
-    maxlen=config['telemetry']['buffer_size'])
-packetBuffers = [event_packet_buffer, telem_packet_buffer]
-# TODO: Use an indexed system so that we have persistent log storage and querying
-packet_lock = Lock()
+from submodules import command_ingest
 
-# FIXME: Rename to telemetry_send_loop()
-def telemetry_send():
-    """
-    Thread method to burst the telemetry every so often
-    """
-    # TODO: check battery levels before sending
-    global telem_packet_buffer, event_packet_buffer
-    time.sleep(60)  # Don't send packets straight away
+# Command Ingest - ability to register commands
+from submodules.command_ingest import command
 
-    while core.get_state() == Mode.NORMAL:
-        # TODO if (adcs.can_TJ_be_seen() == True and len(telem_packet_buffer) + len(event_packet_buffer) > 0):
-        if len(telem_packet_buffer) + len(event_packet_buffer) > 0:
-            telemetry_send_once()
-        time.sleep(config['telemetry']['send_interval'])
+# Log and error classes
+from core import error, log
 
 
-# FIXME: Rename to telemetry_send()
-def telemetry_send_once():
-    """
-    Immediately send telemetry packets in both telemetry and event packet queues
-    """
-    global telem_packet_buffer, event_packet_buffer
-    beg_count = len(telem_packet_buffer) + len(event_packet_buffer)
-    send()
-    logger.debug("Sent " + str(beg_count - len(telem_packet_buffer) -
-                               len(event_packet_buffer)) + " telemetry packets")
+logger = logging.getLogger("TELEMETRY")
 
+general_queue = collections.deque()
```

Comment:
By initializing these variables outside a function they will be reinitialized everytime the module is imported. Put these variables within the `start()` function and initialize them there. 
i.e.
```
def start():
  global general_queue
  general_queue = collections.deque()
```

Reply:
Done in 963480f